### PR TITLE
Add 3d pose inference and visualization

### PR DESCRIPTION
**Pull request by ly015**
This PR added 3d pose inference and visualization interface.

* Add apis/inference_3d.py
* Place 2d/3d keypoint visualization into core/visualization/image.py
* Add show_result method for PoseLifter
* Simplify show_result method of TopDown and BottomUp
* Add a demo for 2d-to-3d pose lifting

### CODE REVIEW

File: demo/body3d_two_stage_img_demo.py [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+from mmpose.core import SimpleCamera
+
+
+def _keypoint_camera_to_world(keypoints,
+                              camera_params,
+                              image_name=None,
+                              dataset='Body3DH36MDataset'):
+    """Project 3D keypoints from the camera space to the world space.
+
+    Args     keypoints (np.ndarray): 3D keypoints in shape [..., 3]
```
Comment (jin-s13) [inline demo/body3d_two_stage_img_demo.py:21]: missing `\n`

File: mmpose/core/visualization/image.py [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+        return img
+
+
+def imshow_keypoints_3d(pose_result,
+                        img=None,
+                        skeleton=None,
+                        pose_kpt_color=None,
+                        pose_limb_color=None,
+                        viz_hight=400,
```
Comment (jin-s13) [inline mmpose/core/visualization/image.py:118]: height

File: mmpose/core/visualization/image.py [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+            - "keypoints_3d" ([K,4]): 3D keypoints
+            - "title" (str): Optional. A string to specify the title of the
+                visualization of this pose result
+        img (str|np.ndarray): Opptional. The image or image path to show input
+            image and/or 2D pose. Note that the image should be given in BGR
+            channel order.
+        skeleton (list of [idx_i,idx_j]): Skeleton described by a list of
+            limbs, each is a pair of joint indices.
+        pose_kpt_color (np.ndarray[Nx3]`): Color of N keypoints. If None, do
+            not nddraw keypoints.
```
Comment (jin-s13) [inline mmpose/core/visualization/image.py:138]: nddraw

Comment (ly015) [reply mmpose/core/visualization/image.py:139]: fixed

File: mmpose/core/visualization/image.py (hunk 2 above)
Comment (ly015) [reply mmpose/core/visualization/image.py]: fixed

File: demo/body3d_two_stage_img_demo.py (hunk 1 above)
Comment (ly015) [reply demo/body3d_two_stage_img_demo.py]: fixed

**Review by jin-s13 — APPROVED** [review]

**Comment by innerlee** [comment]
pyplot is generally slow. I don't know if the speed is okayish

**Comment by ly015** [comment]
Perhaps visualization speed is not a critical issue? We can update it when we have better substitutes for matplotlib.

**Comment by innerlee** [comment]
Yeah it does not block merging. Just saying, my biggest impression on pyplot is its slowness haha.

**Comment by innerlee** [comment]
Unrelated to this pr: I used plotly for 3d drawing once. The speed is nice. https://github.com/plotly/plotly.py

**Pull request merged**